src/main/java/teamdraco/finsandstails/common/items/ArmoredGopjetJetpackItem.java

```diff
@@ -61,36 +62,45 @@ public BlockPos getBlockUnderPlayer(Player player) {
     @Override
     public void onArmorTick(ItemStack stack, Level world, Player player) {
         if (stack.getMaxDamage() - stack.getDamageValue() > 1 || player.isCreative()) {
+            boolean canFly = world.isRainingAt(player.blockPosition());
             int flyingTicksRemaining = 0;
             int stackIndex = -1;
+            BlockPos pos = getBlockUnderPlayer(player);
 
-
-            for (int i = 0; i < player.getInventory().getContainerSize(); i++) {
-                ItemStack inventoryStack = player.getInventory().getItem(i);
-                Item item = inventoryStack.getItem();
-                int ticksJumping = inventoryStack.hasTag() ? inventoryStack.getTag().getInt("FinsFlyingTicks") : 0;
-                if (item == Items.WATER_BUCKET) {
-                    flyingTicksRemaining = 100 - ticksJumping;
-                } else if (item == Items.POTION && PotionUtils.getPotion(inventoryStack) == Potions.WATER) {
-                    flyingTicksRemaining = 30 - ticksJumping;
-                } else {
-                    continue;
-                }
-                stackIndex = i;
-                break;
+            if (!canFly) {
+                if (pos != null)
+                    if (canFly || player.blockPosition().getY() > 0 && world.getBlockState(pos).getMaterial() == Material.WATER) {
```

Comment:
💡 "canFly" seems to be redundant here since the condition above already checks if canFly is false to execute